proof-systems/src/darlin/t_dlog_acc_marlin/mod.rs

```diff
@@ -0,0 +1,734 @@
+use crate::darlin::accumulators::dlog::{DLogItem, DualDLogItem};
+use crate::darlin::t_dlog_acc_marlin::data_structures::{
+    DualSumcheckItem, Proof, ProverKey, SumcheckItem, UniversalSRS, VerifierKey, PC,
+};
+use crate::darlin::t_dlog_acc_marlin::iop::IOP;
+use algebra::{
+    get_best_evaluation_domain, to_bytes, Curve, DensePolynomial,
+    Evaluations as EvaluationsOnDomain, Group, GroupVec, ToBytes,
+};
+use digest::Digest;
+use marlin::iop::Error::VerificationEquationFailed;
+use marlin::iop::LagrangeKernel;
+use marlin::Error;
+use poly_commit::fiat_shamir_rng::{FiatShamirRng, FiatShamirRngSeed};
+use poly_commit::optional_rng::OptionalRng;
+use poly_commit::{
+    evaluate_query_set_to_vec, Evaluations, LabeledCommitment, LabeledPolynomial,
+    LabeledRandomness, PCVerifierKey, PolynomialCommitment,
+};
+use r1cs_core::ConstraintSynthesizer;
+use rand::RngCore;
+use std::marker::PhantomData;
+
+pub mod data_structures;
+pub mod iop;
+#[cfg(test)]
+mod test;
+
+/// A helper struct to bundle the t-dlog accumulator Marlin functions for setup, prove and
+/// verify.
+pub struct TDLogAccMarlin<G1: Curve, G2: Curve, D: Digest>(
+    #[doc(hidden)] PhantomData<G1>,
+    #[doc(hidden)] PhantomData<G2>,
+    #[doc(hidden)] PhantomData<D>,
+);
+
+impl<G1: Curve, G2: Curve, D: Digest + 'static> TDLogAccMarlin<G1, G2, D> {
```

Comment:
As the discussion we had some time ago: we can generalize everything for sure but it's not really worth it at the moment and will lead to other time spent in refactoring rather than implementing core functionalities.
So let's postpone this need of generalizing Marlin/IOP, perhaps I will open an issue about that.